context/app/static/js/components/detailPage/ContributorsTable/utils.ts

```diff
@@ -84,28 +84,50 @@ export const normalizeContributor = (contributor: ContributorAPIResponse): Contr
     ? normalizeCEDARContributor(contributor)
     : normalizeLegacyContributor(contributor);
 
+/**
+ * Given a contributor, determine if they are a contact. Necessary to account
+ *  for different versions of contributors schemas.
+ * @author Austen Money
+ * @param contributor a contributor to be checked.
+ * @returns true if the contributor is a contact, false otherwise.
+ */
+export const contributorIsContact = (contributor: Contributor, contacts: Contributor[]): boolean => {
+  switch (true) {
+    case contributor.isContact:
+      return true;
+    case contacts.find((contact) => contact.orcid === contributor.orcid) !== undefined:
+      return true;
+    default:
+      return false;
+  }
+};
+
 /**
  * Given an array of contributors, sort them by contact status and then alphabetically by name.
  *   Ex sorted array: PI Contact C, PI Contact D, Contact, Contributor A, Contributor B
  * @author Austen Money
  * @param contributors an array of contributors to be sorted.
+ * @param contacts an array of contacts to be used for sorting.
  * @returns a sorted array.
  */
-export const sortContributors = (contributors: Contributor[]): Contributor[] =>
+export const sortContributors = (contributors: Contributor[], contacts: Contributor[]): Contributor[] =>
   contributors.sort((a, b) => {
-    const aIsPIContact = a.isContact && a.isPrincipalInvestigator;
-    const bIsPIContact = b.isContact && b.isPrincipalInvestigator;
+    const aIsContact = contributorIsContact(a, contacts);
```

Comment:
A little bit of shorthand to reduce the logic, you can try `[aIsContact, bIsContact] = [a,b].map((val) => contributorIsContact(val,contacts))`. Not needed here, but good to know!